### ISSUE TITLE
False positive for W312 (Value of an argument is unused)

### ISSUE THREAD

**Issue by jaythomas** [mpeterv/luacheck#96]
The following function trips the error: `value of argument sprite is unused`

```lua
local move_left = function(input, sprite, sprites)
  input.left = true
  if input.right then
    sprite = sprites.actions.default:clone()
  else
    sprite = sprites.actions.move_left:clone()
  end
  return input, sprite
end
```

Although I am never accessing any values in the variable `sprite`, I am certainly mutating it and returning the mutated variable. It doesn't matter if its the first returned value or not. Any advice? Thanks in advance.

**Comment by mpeterv** [comment]
This functions sets the argument variable (mutating would look like `sprite.key = value`) and returns it but it never uses its original value in any way, that's what luacheck complains about. You can pass nil or anything as the second argument and it will have no effect. So this is not a false positive. As a solution you can remove or rename to `_` the argument and use a new local variable inside the function.

**Comment by jaythomas** [comment]
Nice catch! I am definitely not mutating it since I didn't specify what key in `sprite` I wanted. Apologies for the bogus issue report.
